python/tvm/autotvm/task/relay_integration.py

```diff
@@ -122,6 +124,9 @@ def extract_from_multiple_program(mods, params, target, target_host=None, ops=No
 
     env = TaskExtractEnv.get()
 
+    # merge target and target host
+    target, target_host = refresh_host(target, target_host)
```

Comment:
if possible, move this line to the very beginning to each function, so that it could be easier to handle if we want to deprecate target_host in the future

Reply:
Yeah, that's definitely a good idea. Sometimes it appears the target or target host would be updated in the function. Otherwise I can move it forward as much as I can.

Reply:
I have checked every occurance of the function and moved them.